### PR TITLE
Functional tests - Refacto BO - bulk duplicate products

### PR DESCRIPTION
**Pull request by khouloudbelguith**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.0.x
| Description?      | Refacto BO - bulk duplicate products
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | no
| How to test?      | HEADLESS=false TEST_PATH="functional/BO/03*/01*/06*" URL_FO="http://shop.com/" PASSWD="prestashop_demo" npm run test:specific
| Possible impacts? | no

### CODE REVIEW

File: tests/UI/campaigns/functional/BO/03_catalog/01_products/06_bulkActions.js [hunk 1]
```diff
@@ -131,6 +131,31 @@ describe('BO - Catalog - Products : Bulk actions products', async () => {
...
+      const textColumn = await productsPage.getProductNameFromList(page, 1);
+      await expect(textColumn).to.contains('TO DELETE');
+    });
+
+    it('should duplicate products by bulk actions', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'bulkDuplicate', baseContext);
+
+      const duplicateTextResult = await productsPage.duplicateAllProductsWithBulkActions(page);
+      await expect(duplicateTextResult).to.equal(productsPage.productMultiDuplicatedSuccessfulMessage);
+    });
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/03_catalog/01_products/06_bulkActions.js:154]: @khouloudbelguith Shouldn't you check that products are duplicated.

(Number of filtered products are multiplied by 2).

**Review by Progi1984 — COMMENTED** [review]
@khouloudbelguith A small comment ;)

File: tests/UI/campaigns/functional/BO/03_catalog/01_products/06_bulkActions.js (hunk 1 above)
Comment (khouloudbelguith) [reply tests/UI/campaigns/functional/BO/03_catalog/01_products/06_bulkActions.js:154]: Hello @Progi1984,

On this line https://github.com/PrestaShop/PrestaShop/pull/29802/files#diff-42c02b9f7861b0523eb2d3fd7040f04c505be7f89a810f5ee614c4cbf23d7861R155, I will check the exact number of the product after the reset.
Do you think that I need to add the check here also?
Thanks!

Comment (Progi1984) [reply tests/UI/campaigns/functional/BO/03_catalog/01_products/06_bulkActions.js:154]: Oh right. Thanks. I add an another comment.

File: tests/UI/campaigns/functional/BO/03_catalog/01_products/06_bulkActions.js [hunk 2]
```diff
@@ -131,6 +131,31 @@ describe('BO - Catalog - Products : Bulk actions products', async () => {
...
   });
 
+  describe('Bulk duplicate products', async () => {
+    it('should filter products by name', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'filterToBulkDuplicate', baseContext);
+
+      await productsPage.filterProducts(page, 'name', 'TO DELETE');
+
+      const textColumn = await productsPage.getProductNameFromList(page, 1);
+      await expect(textColumn).to.contains('TO DELETE');
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/03_catalog/01_products/06_bulkActions.js:142]: Can't you fetch the number of rows filtered and use it for replacing the `4` in line 155 ?

**Review by matthieu-rolland — COMMENTED** [review]
lgtm, minus @Progi1984 's feedback

**Review by boubkerbribri — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Pull request merged**